embark-ui/src/services/api.js

```diff
@@ -52,6 +52,13 @@ function destroy() {
   return request('delete', ...arguments);
 }
 
+function websocket(credentials, path) {
+  const cnonce = Date.now() + Math.random();
+  const requestHash = hash(cnonce, credentials.token, 'ws', '/', {});
+
+  return new WebSocket(`ws://${credentials.host}/embark-api${path}`, [`${cnonce}|${requestHash}`]);
```

Comment:
I made a conscientious choice to use `|` as a separator here. Turns out there are not a lot of other characters outside `.` (which we need for the `cnonce`) that the protocol header "likes."